troposphere/views/guacamole.py

```diff
@@ -0,0 +1,87 @@
+import json
+import time
+import uuid
+import hmac
+import hashlib
+import base64
+import requests
+import logging
+
+from django.conf import settings
+from django.core.exceptions import PermissionDenied
+from django.http import HttpResponse, HttpResponseRedirect
+
+logger = logging.getLogger(__name__)
+
+guac_server = settings.GUACAMOLE['SERVER_URL']
+secret = settings.GUACAMOLE['SECRET_KEY']
+
+# This function will be called by a POST request made to the URL /guacamole from a button
+# defined in troposphere/static/js/components/projects/resources/instance/details/actions/InstanceActionsAndLinks.jsx
+def guacamole(request):
+
+    if request.user.is_authenticated():
+        logger.info("User is authenticated.")
+
+        # Create UUID for connection ID
+        conn_id = str(uuid.uuid4())
+        base64_conn_id = base64.b64encode(conn_id[2:] + "\0" + 'c' + "\0" + 'hmac')
+
+        # Create timestamp that looks like: 1489181545018
+        timestamp = str(int(round(time.time()*1000)))
+
+        if 'ipAddress' in request.POST and 'protocol' in request.POST:
+
+            # Get IP, protocol, and username from request that was sent from button click
+            ip_address = request.POST['ipAddress']
+            protocol = request.POST['protocol']
+            atmo_username = request.session.get('username','')
+
+            logger.info("User %s initiated %s connection to %s" % (atmo_username, protocol.upper(), ip_address))
+
+            # Change some parameters depending on SSH or VNC protocol
+            # Note: passwd is hardcoded to 'display'. This doesn't seem ideal but it is
+            #       done the same in web_desktop.py
```

Comment:
My issue isn't so much that there is a strange parameter, but that the comment poses a question that someone else will have to answer. If i'm reading this code later, I'll have that same question. `# Note: passwd is hardcoded to 'display'. This seems bad, but it's there because <reason>`